runtime/server/repos.go

```diff
@@ -227,21 +320,23 @@ func (s *Server) UploadMultipartFile(w http.ResponseWriter, req *http.Request, p
 		http.Error(w, fmt.Sprintf("failed to parse file in request: %s", err), http.StatusBadRequest)
 		return
 	}
+	defer f.Close()
 
-	if pathParams["path"] == "" {
+	if path == "" {
 		http.Error(w, "must have a path to file", http.StatusBadRequest)
 		return
 	}
 
-	observability.AddRequestAttributes(ctx,
-		attribute.String("args.instance_id", pathParams["instance_id"]),
-		attribute.String("args.path", pathParams["path"]),
-	)
+	observability.AddRequestAttributes(ctx, attribute.String("args.instance_id", instanceID), attribute.String("args.path", path))
 
-	s.addInstanceRequestAttributes(ctx, pathParams["instance_id"])
+	s.addInstanceRequestAttributes(ctx, instanceID)
 
-	err = s.runtime.PutFile(ctx, pathParams["instance_id"], pathParams["path"], f, true, false)
+	err = s.runtime.PutFile(ctx, instanceID, path, f, true, false)
 	if err != nil {
+		s.logger.Error("failed to write file during upload",
+			zap.String("instanceID", instanceID),
+			zap.String("path", path),
+			zap.Error(err))
 		http.Error(w, fmt.Sprintf("failed to write file: %s", err), http.StatusBadRequest)
 		return
```

Comment:
Not sure this is worth an error log. The most likely error type is request cancellation, which isn't worth an alert.

If you want to improve this, then maybe lean into the middleware that logs response errors by customizing the response status code based on the `err` type? (5xx errors lead to `error`-level logs.)